src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -1147,6 +1147,8 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           assert(!context.inTypeConstructorAllowed, context) //@M
           instantiatePossiblyExpectingUnit(tree, mode, pt)
         }
+        // TODO: we really shouldn't use T* as a first class types (e.g. for repeated case fields), but we can't allow T* to conform to other types (see isCompatible) because that breaks overload resolution
+        else if (isScalaRepeatedParamType(tree.tpe) && !isScalaRepeatedParamType(pt)) adapt(tree setType(repeatedToSeq(tree.tpe)), mode, pt)
```

Comment:
This looks okay on the assumption that the `*` only appears in the outermost part of `tree.tp` / `pt`.

For case classes, where we have the unfortunate case of `*` appearing at all, this seems to hold true:

```
scala> class C(val c: String*)
defined class C

scala> new C("")
res0: C = C@558756be

scala> res0.c
   val c: String*

scala> res0.c
res1: Seq[String] = WrappedArray("")

scala> class C(val c: String*) { def foo = Seq(c) }
defined class C

scala> new C("").foo
   def foo: Seq[Seq[String]]

scala> { () => new C("").foo }
res3: () => Seq[Seq[String]] = $$Lambda$1467/1307655632@2ed7978c
```

Reply:
Yep, we shouldn't emit fields with type T*, but then we need to adjust other logic to turn those back into varargs where needed. I chose the expedient route :-)